enterprise/views.py

```diff
@@ -1659,7 +1660,11 @@ def get_course_run_id(enterprise_customer, course_key):
         except ImproperlyConfigured:
             raise Http404
 
-        course_run = get_current_course_run(course)
+        users_active_course_runs = [
+            course_run for course_run in course['course_runs']
+            if EnrollmentApiClient().is_enrolled(user.username, course_run['key'])
```

Comment:
I don't love making this API call inside a loop, we run the risk of rate limiting ourselves if a course has many runs. Can we update this to either:
- make a single call with the course and have the enrollment API endpoint do the lookup of course_runs
- have the endpoint take the full list of runs to validate enrollment
- change it to call the enrollment code directly instead of via API

I don't have a strong preference on which of those 3 approaches (there may already be API endpoints for one of the first 2 options, I haven't looked) but I think we should update this before merging

Reply:
@georgebabey I have updated the code. I noticed that there is an endpoint which returns the all active enrollment of users. I have introduced the new function in utils to filter. Let me know if you have still concerns.